packages/environment/develop.js

```diff
@@ -116,20 +140,31 @@ const Develop = {
     });
   },
 
-  connectOrStart: async function (options, ganacheOptions) {
-    options.retry = false;
-
-    const ipcNetwork = options.network || "develop";
+  /**
+   * Connect to a managed Ganache service. This will connect to an existing
+   * Ganache service if one exists, or, create a new one to connect to.
+   *
+   * @param {Object} ipcOptions
+   * @param {string} ipcOptions.network the network name.
+   * @param {Object} ganacheOptions to be used if starting a Ganache service is
+   *        necessary.
+   * @param {TruffleConfig} truffleConfig the truffle config.
```

Comment:
I'm not super keen on adding config as an argument here... could we, possibly as follow-on work, just pull out the specific fields we want instead?

Reply:
Yep, added #5784 for that.